ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java

```diff
@@ -182,6 +185,65 @@ private ReactContext getReactContext() {
   protected void onLayout(
       boolean changed, int textViewLeft, int textViewTop, int textViewRight, int textViewBottom) {
     // TODO T62882314: Delete this method when Fabric is fully released in OSS
+    Spanned text = (Spanned) getText();
+    Layout layout = getLayout();
+    TextInlineViewPlaceholderSpan[] placeholders =
+        text.getSpans(0, text.length(), TextInlineViewPlaceholderSpan.class);
+    HashMap<Integer, Integer> imageLineHeights = new HashMap<Integer, Integer>();
+    for (TextInlineViewPlaceholderSpan placeholder : placeholders) {
+      int height = placeholder.getHeight();
+      int start = text.getSpanStart(placeholder);
+      int line = layout.getLineForOffset(start);
+      if (imageLineHeights.get(line) != null && imageLineHeights.get(line) > height) {
+        imageLineHeights.put(line, height);
+      }
+      if (imageLineHeights.get(line) == null) {
+        imageLineHeights.put(line, height);
+      }
+    }
+    // loop over each CustomLineHeightSpan
+    // use getHeight to retrieve height of that line
+    // save it and later retrieve it
+    Integer highestLineHeight = -1;
+    if (getText() instanceof Spanned) {
+      CustomLineHeightSpan[] spans = text.getSpans(0, text.length(), CustomLineHeightSpan.class);
+      if (layout != null) {
+        if (spans.length != 0) {
+          for (CustomLineHeightSpan span : spans) {
+            if (highestLineHeight == 0f || span.getLineHeight() > highestLineHeight) {
+              highestLineHeight = span.getLineHeight();
+            }
+          }
+        }
+      }
```

Comment:
Retrieve lineHeight from ReactAlignSpan instead of looping over each CustomLineHeightSpan